### PR TITLE
fix(search): stop event propagation for characters and numbers

### PR DESCRIPTION
**Pull request by sjohnson-sage**
Fixes: #3830

### Proposed behaviour
Stop event propagation when typing characters or numbers into the `Search` input, these should not propagate and affect keyboard nav etc of components further up the tree

### Current behaviour
Error when typing into `Search` when it's inside of a submenu because the event bubbles up to the keyboard nav in `Submenu` when it shouldn't

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### Testing instructions
Use the sandbox from the attached Github issue and check the console on http://localhost:9001/?path=/story/design-system-menu--submenu-with-search

### CODE REVIEW

File: src/__experimental__/components/search/search.spec.js [hunk 1]
```diff
@@ -472,4 +476,48 @@ describe("Search", () => {
...
   });
+
+  describe("when typing into the input", () => {
+    const stopPropagationFn = jest.fn();
+
+    afterEach(() => {
+      jest.clearAllMocks();
+    });
+
+    it("should stop propagation of the event for character keys", () => {
```
Comment (dawidzarzycki) [inline src/__experimental__/components/search/search.spec.js:487]: I think that we can create a `it.each` for these examples

Comment (sjohnson-sage) [reply src/__experimental__/components/search/search.spec.js:487]: I'm not sure there's any need for that here really, we only have 2 tests making the same assertion, the third test is asserting the opposite

Comment (dawidzarzycki) [reply src/__experimental__/components/search/search.spec.js:487]: You are right! I thought 3 of them are about the same :)
Ignore my comment then

**Review by dawidzarzycki — DISMISSED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by edleeks87 — DISMISSED** [review]

**Review by dawidzarzycki — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by KatarzynaQA — APPROVED** [review]

**Pull request merged**